### DISCUSSION TITLE
Valetudo 2021.11.1

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1223] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2021.11.1</h2></p>
</div>

Bugfixes, polishing, two minor breaking changes and new features.
Don't forget: If you're running Valetudo 2021.11.0, you can now use the integrated updater :)

### Polishing

- Controls in the UI were slightly rearranged/reworked.
- The mobile controls drawer now works slightly different and more importantly more reliably with better performance (+ 50kb less bundle size).
- There's a neat loading splashscreen, which you should only see for longer than a second if your Wi-Fi sucks

![image](https://user-images.githubusercontent.com/974410/142760316-4af4fd83-79dd-4c42-857f-fa71ef2d2bc2.png) ![image](https://user-images.githubusercontent.com/974410/142760318-2b74c894-0572-4a55-8c71-6a655660f081.png)

### Provisioning

The UI now checks if the robot is connected to a Wi-Fi network and if not opens a special provisioning page.
It's pretty neat and also fixes the issue of Dreames not being provisionable with the new UI.

![image](https://user-images.githubusercontent.com/974410/142760331-ee5a4031-c692-49be-9ad8-4144f35bb5e0.png)

### CurrentStatisticsCapability

Statistics related to the current (or most recent) operation of the robot are now back.
Yes, they are also being published to MQTT. Stop asking.

Data is published to MQTT as machine-readable (seconds for runtime, cm² for the area) as it is meant to build automations with it.
Using cm² instead of m² saves us from having to deal with floats.

### TotalStatisticsCapability

More importantly, you can now get total statistics from your robot. 

Depending on your model those can be

- Total operation count
- Total runtime
- Total area

This isn't available via the UI yet as it is yet TBD how to do that.
Maybe have like badges, achievements or a rank system?
Maybe have random trivia related to the numbers? (e.g. having cleaned 0.0001% of the Saarland)

Feel free to leave your input in the comments. ~~Like, subscribe and click the bell icon~~
If you're good with graphics design, something like e.g. a Corporal IV vacuum robot badge would be a great contribution.

## Zeroconf hostname changes

By actually reading the relevant RFCs, we've discovered that `valetudo_something.local` is not a valid bonjour hostname as the `_` is illegal.
While it does seem to work in most setups, it might cause problems in some, which is why starting with this release, Valetudo will use an RFC-compliant hostname: `valetudo-something.local`.

Please **update your bookmarks** if you were using any.

### Home Assistant 2021.11 MQTT Changes

Valetudo 2021.11.1 makes use of new features introduced in Home Assistant 2021.11 (namely `configuration_url` and `entity_category`).
This means that **Home Assistant >= 2021.11 is now required** if you want to use it with Valetudo.

You might also have to delete your Valetudo device in HA and let it get rediscovered so that the changes are applied properly.

It now looks like this:

![image](https://user-images.githubusercontent.com/974410/142760339-aa5b2314-692d-4167-9689-cec21f718924.png)

### Possible MQTT OOM fixes

We have received unconfirmed, unsubstantiated and possibly falsified reports of Valetudo going OOM related to Wi-Fi connectivity issues and the MQTT interface.

To combat this, Valetudo now drops any outgoing MQTT message if there are more than 1 MiB of outgoing messages buffered. This may or may not ever happen.
If you see a warning in your log related to this, please report back. I have not been able to reproduce this.

### Homie Consumables

Home consumable durations are now reported as an int. This should fix issues with OpenHab.

### Dreame paths

Dreame paths now parse to multiple path map entities instead of a single one, which fixes the visual with lines going straight across the map connecting the previous path to the next one.
Consumers of the Valetudo map data might have to be updated to properly display multiple path entities.

### 1C-related fixes

I've bought a Dreame 1C to do some QA, which led to a few fixes.
I'm now quite happy with the current state of its implementation in Valetudo.

### Features

* **core:** CurrentStatisticsCapability ([8e4e183](https://github.com/Hypfer/Valetudo/commit/8e4e1839c217cc081f3172f227e935c2dc616bf1))
* **core:** CurrentStatisticsCapability & TotalStatisticsCapability properties ([4fc9df2](https://github.com/Hypfer/Valetudo/commit/4fc9df271af4dbb9a9c1841d5f23a292d8cd0af3))
* **core:** Introduce ConsumableMonitoringCapability properties and remove mqtt consumables hack ([a998ee1](https://github.com/Hypfer/Valetudo/commit/a998ee1eeeba2fd808c797bed719933d85922252))
* **core:** TotalStatisticsCapability ([55b6696](https://github.com/Hypfer/Valetudo/commit/55b6696ecfb7f6d86b34da2e9e37b07287f8ef68))
* **MockRobot:** Add MockCurrentStatisticsCapability + fix MockConsumableMonitoringCapability ([bda0b1a](https://github.com/Hypfer/Valetudo/commit/bda0b1adc54c8b651a3d23db8a4dc4638b460e48))
* **mqtt:** Homie consumables should always be int ([c39ad93](https://github.com/Hypfer/Valetudo/commit/c39ad9380fc1d4cf752a3d95f0f57512e608c64f))
* **mqtt:** Limit outgoing message buffer size to prevent OOM situations ([1a74f91](https://github.com/Hypfer/Valetudo/commit/1a74f9114de877f01b728e281b843e8459100b0f))
* **mqtt:** Publish current statistics to mqtt ([102a078](https://github.com/Hypfer/Valetudo/commit/102a07890fe0a13efbda8b568b177fa4940478be))
* **ui:** Add Valetudo loading splash ([a7ca33c](https://github.com/Hypfer/Valetudo/commit/a7ca33c845fa1e32d1a217115e48cd534c38b7bb))
* **ui:** Collapse preset selection controls ([4f8aaec](https://github.com/Hypfer/Valetudo/commit/4f8aaec2f43b9e1a3bb21df88083b80509652fde))
* **ui:** Display current statistics ([0fbc182](https://github.com/Hypfer/Valetudo/commit/0fbc182b0998cf1b3804b52ef24ee5d2d14c1b21))
* **ui:** Slightly decrease required zoom level for segment label text ([08209a3](https://github.com/Hypfer/Valetudo/commit/08209a3164156db6e5e3ba340342bfbbf75bd7e3))
* **vendor.dreame:** Total and current statistics for the 1C ([c1a51e9](https://github.com/Hypfer/Valetudo/commit/c1a51e916a9d123fcd6f9621eff02b645f79d702))
* Provision Wizard thingy ([a6f6ded](https://github.com/Hypfer/Valetudo/commit/a6f6dedc851d6852e7b25ba1dc5222f3130ef210))
* **ui:** Reorder state display ([828e54d](https://github.com/Hypfer/Valetudo/commit/828e54da624d2e00f500499e3497867fdb2ad6eb))
* **ui:** Replace bottom sheet with simpler mobile controls ([17ff7ec](https://github.com/Hypfer/Valetudo/commit/17ff7ecb1105d9904f382a15b92dd5a6b988dd1f))
* **vendor.dreame:** DreameCurrentStatisticsCapability ([656a19e](https://github.com/Hypfer/Valetudo/commit/656a19ee39f1411bb962e61077486d9336dd79b5))
* **vendor.dreame:** DreameTotalStatisticsCapability ([4bb9342](https://github.com/Hypfer/Valetudo/commit/4bb934208ff52ad7fc1f42b82e5046362353ab6a))
* **vendor.roborock:** RoborockCurrentStatisticsCapability ([93aa77b](https://github.com/Hypfer/Valetudo/commit/93aa77b09b7924baf5fce594d6455f4a5f51837e))
* **vendor.roborock:** RoborockKeyLockCapability ([#1222](https://github.com/Hypfer/Valetudo/issues/1222)) ([02e155c](https://github.com/Hypfer/Valetudo/commit/02e155cec954cc75e41c5fcbf8ebe37acffb4c37))
* **vendor.roborock:** RoborockTotalStatisticsCapability ([e0cbbff](https://github.com/Hypfer/Valetudo/commit/e0cbbff7470426119dd1eff6bde064a9387209b5))
* **vendor.viomi:** ViomiCurrentStatisticsCapability ([9dd1091](https://github.com/Hypfer/Valetudo/commit/9dd1091cbc7a63ac7c0f61b40bc73abd026572bc))
* **webserver:** Add 404 page ([8f5095c](https://github.com/Hypfer/Valetudo/commit/8f5095cac124c0ee0a6b8cdece07223c3374c51f))

### Bug Fixes

* **mqtt:** Attempt to mitigate problems related to connectivity issues ([cf0f2d6](https://github.com/Hypfer/Valetudo/commit/cf0f2d691017ac69635d17f04062c260ebf62091))
* **mqtt:** homie.durationsAsInteger should not be a breaking change ([bbd918a](https://github.com/Hypfer/Valetudo/commit/bbd918ac35530c0a7a8e8b684ebbe05967506bfc))
* **ui:** Align preset selection labels with icons ([c6803f6](https://github.com/Hypfer/Valetudo/commit/c6803f6e2d5a60172d945bd62ab47badd679d234))
* **ui:** Attachment buttons should be disabled ([922572a](https://github.com/Hypfer/Valetudo/commit/922572a779e0e1402ca53bb3e9b8a1c82605ea70))
* **ui:** Fix CurrentStatistics trying to map something that isn't an array ([4d6f3d8](https://github.com/Hypfer/Valetudo/commit/4d6f3d869202288fb48bc50ee38d124c611fac78))
* **ui:** Fix mobile control scrolling ([6264400](https://github.com/Hypfer/Valetudo/commit/6264400a473b36aff9f5409a5e157de536389e6d))
* **ui:** Only check for provisioning once ([6bd0480](https://github.com/Hypfer/Valetudo/commit/6bd0480d8d40600aa5f438d060a21b2ba775925a))
* **ui:** Only regular mouse clicks should interact with the map ([d452d88](https://github.com/Hypfer/Valetudo/commit/d452d88206afafbf4017138fa8b52edd38f1f70d))
* **ui:** Preset selection labels should not be user selectable ([965a061](https://github.com/Hypfer/Valetudo/commit/965a0610c2dd73e028f944ed3a49a5e3c507d7e6))
* **ui:** Swap presetSelection open/close icons ([380109b](https://github.com/Hypfer/Valetudo/commit/380109baa7bc6abfb06169eaeac036a7235af005))
* **ui:** The whole controls body should not be user-selectable ([861f15f](https://github.com/Hypfer/Valetudo/commit/861f15f6da3633c266f0fbca8471193642b8ae57))
* **vendor.dreame:** 1C shall ignore persistent map property updates ([5744346](https://github.com/Hypfer/Valetudo/commit/5744346272bf443ca734da77c5af938a559d314b))
* **vendor.dreame:** Assume that we're in the docked state if we're charging on the 1C ([c73e062](https://github.com/Hypfer/Valetudo/commit/c73e06207c098deae3e596de135c5e2cbf50f4da))
* **vendor.dreame:** Fine-tune 1C manual control ([72614e2](https://github.com/Hypfer/Valetudo/commit/72614e29977297078cf37425be361ad5310aad07))
* **vendor.dreame:** Fine-tune manual control ([cc267c7](https://github.com/Hypfer/Valetudo/commit/cc267c75fe02178343cbddd43099ff59cfe8b16f))
* **vendor.dreame:** Fix Dreame1CManualControlCapability ([9c85ca5](https://github.com/Hypfer/Valetudo/commit/9c85ca5f8c09dc5e1371f4b805767a6bec54175c))
* **vendor.dreame:** Properly parse path data to multiple paths ([e083af2](https://github.com/Hypfer/Valetudo/commit/e083af2759f37706cda1e45a0b083b0782112f3b))
* **vendor.dreame:** Remove capabilities for models that don't support them ([0d6d6ad](https://github.com/Hypfer/Valetudo/commit/0d6d6ad3ad44bdadb9ceea830652e8e742487776))
* **vendor.dreame:** The 1C does not support multiple iterations for segment cleanups ([2d35bd9](https://github.com/Hypfer/Valetudo/commit/2d35bd9a0f00f52f9e0a67dae5f31d50e7a87cb3))
* **vendor.viomi:** Fix ViomiPersistentMapControlCapability ([#1218](https://github.com/Hypfer/Valetudo/issues/1218)) ([80d7bb2](https://github.com/Hypfer/Valetudo/commit/80d7bb2bd91377d1c06a0c6afa4a262d0b7511e7))
* Fix LinuxWifiConfigurationCapability not_connected state reporting ([25271f2](https://github.com/Hypfer/Valetudo/commit/25271f24c95976bae9b201ff1a6d2625c3e643a1))
* **vendor.roborock:** RoborockSpeakerVolumeControlCapability.getVolume() should return a number ([32898d1](https://github.com/Hypfer/Valetudo/commit/32898d1cd27b3498f00e6dd4d67c2de09d0ec1ed))
* **vendor.viomi:** Fix volume control ([0269182](https://github.com/Hypfer/Valetudo/commit/0269182f5e94d055d908b9c16e1524ef5a9f949d))
* **webserver:** try/catch for SpeakerVolumeControlCapabilityRouter ([5cb7c5a](https://github.com/Hypfer/Valetudo/commit/5cb7c5ac3c09c9138ac11316aff3c7dcfa22fde1))
* Nested assign configuration defaults + don't persist if there are no changes to persist ([b12c452](https://github.com/Hypfer/Valetudo/commit/b12c452b4d693326d0ebec81bf6b1145c8ca9de5))

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2021.11.1'>Valetudo 2021.11.1</a>.</em>

**Comment by StefanSchoof** [comment]
The first update, where I used the Updater and it worked for me without any issues. I like it.

**Reply by nefh** [reply]
The same here. Whole update process really looked great, I did not expect i.e. displaying whole changelog. Great idea!

**Comment by DaanOlbrechts** [comment]
Updater worked for me as well! Great!

**Comment by gvolt** [comment]
Successfully updated using the updater.

Thanks for refining the UI and also for re-adding statistics. It's nice to be able to check how long a certain cleaning session took and how big the cleaned area was.

**Comment by Skeletorjus** [comment]
Great release! Very nice to have statistics back!
They do seem to be missing units (seconds for time and cm² for area) in the HA MQTT autodiscovery.
Not a huge issue, but thought I'd mention it.

Updater worked great :)

**Reply by Hypfer** [reply]
Oh. That explains why it looked wrong. I just wasn't able to tell what exactly was off

Fixed: 7f052dc3

**Reply by gvolt** [reply]
Haha, I didn't perceive the missing units either. That's the power of imagination: You see what you think you see. :)

**Comment by rimrul** [comment]
Regarding the new provisioning page, there have been reports of users intentionally running their robot unprovisioned in the past. 

One such example I found is https://github.com/Hypfer/Valetudo/discussions/1053

While it was certainly never recommended, this has worked to a certain degree before. 

The screenshots don't seem to have a way to leave the provisioning page without provisioning. Is that something we should consider adding?

**Reply by Hypfer** [reply]
Hm that unfortunately never worked quite right as the cloud interface was never working and therefore there were no maps.
It also only works for now EOL roborock models.

I think it's reasonable to expect people to use the robot in a provisioned state. They don't even need Internet for that
Something like the `TP-Link TL-WR802N N300 WLAN Nano Router` for 10-20€ should be sufficient to get everything working.
It also uses basically zero power
